src/projects/mesonproject.vala

```diff
@@ -783,6 +795,38 @@ class Vls.MesonProject : Project {
 
     public MesonProject (string root_path, FileCache file_cache, Cancellable? cancellable = null) throws Error {
         base (root_path, file_cache);
+        // read system envionment;
+        string[] system_env_key = Environment.list_variables();
+        if (system_env_key != null) {
+            foreach(var item in system_env_key) {
+                var sys_env = item+"="+Environment.get_variable(item);
+                debug("system env is %s", sys_env);
+                special_envp += sys_env;
+            }
+        }
+        // get config from settings.json
+        var check_path = root_path + "/.vscode/settings.json";
+        warning("meson check option at %s", check_path);
+        var settings_json = File.new_for_path(check_path);
+        if (settings_json.query_exists()) {
+            var dis = new DataInputStream(settings_json.read());
+            var settings_json_parser = new Json.Parser();
+            settings_json_parser.load_from_stream(dis);
+            var root_node = settings_json_parser.get_root();
+            var root_object = root_node.get_object();
+            var ret_options = root_object.get_array_member("mesonbuild.configureOptions");
+            foreach(var option in ret_options.get_elements()) {
+                var opt = option.get_string();
+                special_args += opt;
+            }
+            var compile_options = root_object.get_array_member("mesonbuild.compileOptions");
+            foreach(var option in compile_options.get_elements()) {
+                var opt = option.get_string();
+                special_comp += opt;
+            }
+            // maybe can reuse mesonbuild buildDir
+            // this.build_dir = root_object.get_string_member_with_default("mesonbuild.buildFolder", root_path +"/builddir");
```

Comment:
Does this not work, or why is it commented out? I think it would be useful.

Reply:
> Does this not work, or why is it commented out? I think it would be useful.

This has been tested and I will open it later